security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -179,7 +181,43 @@ func (sf *SecretFetcher) scrtDeleted(obj interface{}) {
 	log.Debugf("secret %s is deleted", scrt.GetName())
 }
 
-// FindIngressGatewaySecret returns the secret for a k8sKey, or empty secret if no
+func (sf *SecretFetcher) scrtUpdated(oldObj, newObj interface{}) {
+	oscrt, ok := oldObj.(*v1.Secret)
+	if !ok {
+		log.Warnf("Failed to convert to old secret object: %v", oldObj)
+		return
+	}
+	nscrt, ok := newObj.(*v1.Secret)
+	if !ok {
+		log.Warnf("Failed to convert to new secret object: %v", newObj)
+		return
+	}
+
+	okey := oscrt.GetName()
+	nkey := nscrt.GetName()
+	if okey != nkey {
+		log.Warnf("Failed to update secret: name does not match (%s vs %s).", okey, nkey)
+		return
+	}
+	if bytes.Equal(oscrt.Data[ScrtCert], nscrt.Data[ScrtCert]) && bytes.Equal(oscrt.Data[ScrtKey], nscrt.Data[ScrtKey]) {
+		log.Debugf("secret %s does not change, skip update", okey)
+		return
+	}
+	sf.secrets.Delete(okey)
+
+	t := time.Now()
+	ns := &model.SecretItem{
+		ResourceName:     nkey,
+		CertificateChain: nscrt.Data[ScrtCert],
+		PrivateKey:       nscrt.Data[ScrtKey],
+		CreatedTime:      t,
+		Version:          t.String(),
+	}
+	sf.secrets.Store(nkey, *ns)
+	log.Debugf("secret %s is updated", nscrt.GetName())
```

Comment:
you may need to notify client that secret is changed by invoke notifyCallback like
```
					if sc.notifyCallback != nil {
						// Send the notification to close the stream connection if both cert and token have expired.
						if err := sc.notifyCallback(key.ProxyID, key.ResourceName, nil /*nil indicates close the streaming connection to proxy*/); err != nil {
							log.Errorf("Failed to notify for proxy %q: %v", proxyID, err)
						}
					} else {
						log.Warnf("secret cache notify callback isn't set")
					}

```